pkg/util/node/node.go

```diff
@@ -167,16 +167,22 @@ func PatchNodeCIDR(c clientset.Interface, node types.NodeName, cidr string) erro
 
 // PatchNodeStatus patches node status.
 func PatchNodeStatus(c v1core.CoreV1Interface, nodeName types.NodeName, oldNode *v1.Node, newNode *v1.Node) (*v1.Node, error) {
-	oldData, err := json.Marshal(oldNode)
-	if err != nil {
-		return nil, fmt.Errorf("failed to marshal old node %#v for node %q: %v", oldNode, nodeName, err)
-	}
 
 	// Reset spec to make sure only patch for Status or ObjectMeta is generated.
 	// Note that we don't reset ObjectMeta here, because:
 	// 1. This aligns with Nodes().UpdateStatus().
 	// 2. Some component does use this to update node annotations.
 	newNode.Spec = oldNode.Spec
+
+	return PatchNode(c,nodeName,oldNode,newNode)
+}
+
+func PatchNode(c v1core.CoreV1Interface, nodeName types.NodeName, oldNode *v1.Node, newNode *v1.Node) (*v1.Node, error) {
```

Comment:
What was the reason for the new method?

I think we should leave this how it was since `PatchNode` has side effects since it also patches the status subresource. That shouldn't be something we do all the time.

Reply:
It's to avoid duplicating code. If you want I'll add a comment to explicit that

Reply:
I think we should keep it as it was. Nothing new is calling that method and patching the status shouldn't be standard. That means when someone else wants to use the common utility they will need a new method.

Reply:
I can buy that.

It'll add a bit of patch complexity on the `#addcloudNode()` on the other hand.

Reply:
Hrm, so you could keep `PatchNode` and just pullout the common functionality related to creating the patch bytes that way `PatchNodeStatus` and `PatchNode` aren't all copy/paste. Something like
```
func getPatchBytes(nodeName types.NodeName, oldNode *v1.Node, newNode *v1.Node) ([]byte, error) {
	oldData, err := json.Marshal(oldNode)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal old node %#v for node %q: %v", oldNode, nodeName, err)
	}

	newData, err := json.Marshal(newNode)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal new node %#v for node %q: %v", newNode, nodeName, err)
	}

	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1.Node{})
	if err != nil {
		return nil, fmt.Errorf("failed to create patch for node %q: %v", nodeName, err)
	}
	return patchBytes, err
}
```

Thoughts @yastij ?

Reply:
SGTM